validation/l2oo_test.go

```diff
@@ -0,0 +1,159 @@
+			checkEquality(desired.StartingTimestamp, actual.StartingTimestamp),
+			incorrectMsg("startingTimestamp", desired.StartingTimestamp, actual.StartingTimestamp))
+		require.Condition(t,
+			checkEquality(desired.SubmissionInterval, actual.SubmissionInterval),
+			incorrectMsg("submissionInterval", desired.SubmissionInterval, actual.SubmissionInterval))
+		require.Condition(t,
+			checkEquality(desired.L2BlockTime, actual.L2BlockTime),
+			incorrectMsg("l2BlockTime", desired.L2BlockTime, actual.L2BlockTime))
+		require.Condition(t,
+			checkEquality(desired.FinalizationPeriodSeconds, actual.FinalizationPeriodSeconds),
+			incorrectMsg("finalizationPeriodSeconds", desired.FinalizationPeriodSeconds, actual.FinalizationPeriodSeconds))
+	}
+
+	checkL2OOParams := func(t *testing.T, chain *ChainConfig) {
+		rpcEndpoint := Superchains[chain.Superchain].Config.L1.PublicRPC
+
+		require.NotEmpty(t, rpcEndpoint)
+
+		client, err := ethclient.Dial(rpcEndpoint)
+		require.NoErrorf(t, err, "could not dial rpc endpoint %s", rpcEndpoint)
+
+		contractAddress, err := Addresses[chain.ChainID].AddressFor("L2OutputOracleProxy")
+		require.NoError(t, err)
+
+		var desiredParams L2OOParams
+		switch chain.Superchain {
+		case "mainnet":
+			desiredParams = OPMainnetL2OOParams
+		case "goerli":
+			desiredParams = OPGoerliL2OOParams
+		case "sepolia":
+			desiredParams = OPSepoliaL2OOParams
+		case "goerli-dev-0":
+			desiredParams = OPGoerliDev0L2OOParams
+		case "sepolia-dev-0":
+			desiredParams = OPSepoliaDev0L2OOParams
+		default:
+			t.Fatalf("superchain not recognized: %s", chain.Superchain)
+		}
+
+		actualParams, err := getl2OOParamsWithRetries(context.Background(), common.Address(contractAddress), client)
+		require.NoErrorf(t, err, "RPC endpoint %s", rpcEndpoint)
+
+		requireEqualParams(t, desiredParams, actualParams)
+
+		t.Logf("L2OutputOracle config params acceptable")
+
+	}
+
+	for chainID, chain := range OPChains {
+		if !isExcluded[chainID] {
+			t.Run(chain.Name+fmt.Sprintf(" (%d)", chainID), func(t *testing.T) { checkL2OOParams(t, chain) })
+		}
+	}
+}
+
+// getResourceMeteringwill gets each of the parameters from the L2OutputOracle at l2OOAddr,
+// retrying up to 10 times with exponential backoff.
+func getl2OOParamsWithRetries(ctx context.Context, l2OOAddr common.Address, client *ethclient.Client) (L2OOParams, error) {
+	const maxAttempts = 3
+	l2OO, err := bindings.NewL2OutputOracle(l2OOAddr, client)
+	if err != nil {
+		return L2OOParams{}, err
+	}
+
+	params := L2OOParams{}
+	params.StartingBlockNumber, err = retry.Do(ctx, maxAttempts, retry.Exponential(),
+		func() (*big.Int, error) {
+			return l2OO.StartingBlockNumber(&bind.CallOpts{Context: ctx})
+		})
+	if err != nil {
+		return L2OOParams{}, fmt.Errorf("could not get startingBlockNumber: %w", err)
+	}
+	params.StartingTimestamp, err = retry.Do(ctx, maxAttempts, retry.Exponential(),
+		func() (*big.Int, error) {
+			return l2OO.StartingTimestamp(&bind.CallOpts{Context: ctx})
```

Comment:
[nit] You can reuse the callopts

Reply:
[fce979a](https://github.com/ethereum-optimism/superchain-registry/pull/79/commits/fce979a3da3f98b6f90914c5425208b0858481d2)